crates/router/src/core/webhooks/outgoing.rs

```diff
@@ -733,7 +750,7 @@ async fn api_client_error_handler(
         .change_context(errors::WebhooksFlowError::OutgoingWebhookRetrySchedulingFailed)?;
     }
 
-    Err(error)
+    Ok(event)
```

Comment:
Let's not change the return value here, we'd want this function to return an error itself.

Let me clarify:

We want the `trigger_webhook_to_merchant()` function to return errors in case the client faces an error when sending the webhook (i.e., `api_client_error_handler()`), or if the merchant server returns a non-2xx status code (i.e., `error_response_handler()`). This is because when we are firing the analytics event, we check if the `trigger_webhook_to_merchant()` function returned an error, and mark the webhook delivery attempt as unsuccessful (in the ClickHouse event).

Keeping this is mind, we cannot return the updated event when the merchant server returns a non-2xx status code, because `error_response_handler()` returns an error which is propagated by the `?`, and we just cannot return the updated event after that function call. The same applies to the `api_client_error_handler()` invocation as well. This requires us to fetch the updated event in the `raise_webhooks_analytics_event()` function.

So, let's revert all changes to the `trigger_webhook_to_merchant()` and `api_client_error_handler()` functions, and update the `raise_webhooks_analytics_event()` function to fetch the updated event from the database with the merchant ID and event ID, and populate the fields in the analytics event from this updated event object.

My bad, I should have caught this during my previous review.

Reply:
Got it, will revert the changes and fetch the updated even from db.